CMakeLists.txt

```diff
@@ -10,7 +16,29 @@ include(GNUInstallDirs)
 include(BuildPlatformDetection)
 include(SetProjectDependencies)
 
-set(PYTHON_EXECUTABLE ${CMAKE_INSTALL_PREFIX}/bin/python3)
+# For all OSs:
+if($ENV{PYTHONPATH})
+        set(ENV{PYTHONPATH} ${CMAKE_INSTALL_PREFIX}/lib/python3/dist-packages/:${CMAKE_INSTALL_PREFIX}/lib/python3.5:${CMAKE_INSTALL_PREFIX}/lib/python3.5/site-packages:$ENV{PYTHONPATH})
+    else()
+        set(ENV{PYTHONPATH} ${CMAKE_INSTALL_PREFIX}/lib/python3/dist-packages/:${CMAKE_INSTALL_PREFIX}/lib/python3.5:${CMAKE_INSTALL_PREFIX}/lib/python3.5/site-packages)
+    endif()
+
+# OS-specific changes
+if(BUILD_OS_LINUX)
+    # Preparing for building into a prefix on Linux
+    set(ENV{PATH} ${CMAKE_INSTALL_PREFIX}/bin/:$ENV{PATH} CACHE STRING "PATH" FORCE)
+    set(ENV{LD_LIBRARY_PATH} ${CMAKE_INSTALL_PREFIX}/lib/)
+    set(PYTHON_EXECUTABLE python3)
+    set(PYTHON_EXECUTABLE_PREFIXED LD_LIBRARY_PATH=$ENV{LD_LIBRARY_PATH} PATH=$ENV{PATH} PYTHONPATH=$ENV{PYTHONPATH} ${PYTHON_EXECUTABLE})
+elseif(BUILD_OS_WINDOWS)
+    # Minimalistic preinstalled Python on Windows:
+    set(PYTHON_EXECUTABLE python)
```

Comment:
On Windows, it is probably better to use find_package(PythonInterp) if we want to use a system Python. 

However, the intention of this project is to build _all_ dependencies. This includes Python, even on Windows. If someone wants to use system packages, they should use cura-build without cura-build-environment. Currently the only thing that we should fix in that regard is providing binaries for libArcus somewhere, which are probably simpler to produce by just compiling protobuf + arcus separately than special-casing all of cura-build-environment.